src/component/ConsentPreferencesDashboard.tsx

```diff
@@ -252,7 +252,7 @@ export class ConsentPreferencesDashboard extends Component<Props, State> {
                 showCmp();
             })
             .catch(error => {
-                // TODO: ERROR HANDLING
+                // TODO: Handle error by removing the component
                 // eslint-disable-next-line no-console
                 console.log('Error', error);
```

Comment:
The `<ConsentManagementPlatform>` component (one level up) takes a prop `onClose` which is a function. We execute this when we close the CMP. What do you think to feeding this into this `<ConsentPreferencesDashboard.tsx>` component as a part of a prop eg. `onVendorListError`. We could execute `onVendorListError` within this catch block. eg.

```js
// ConsentManagementPlatform.tsx
<ConsentPreferencesDashboard.tsx onVendorListError{() => {
  this.props.onClose();
}} />
```

Reply:
Yeah sound good. Will make this change also.

Reply:
Looking at our code it seems like calling `hideCmp()` would achieve the same without adding an extra prop. I've made that change but was wondering what you think about it. Happy to go a different way if you think it's better.

Reply:
Happy to go with `hideCmp` in this case